src/controllers/events.cr

```diff
@@ -560,9 +563,14 @@ class Events < Application
     raise Error::NotFound.new("event #{event_id} not found on system calendar #{cal_id}") unless event
 
     # ensure we have the host event details
-    if client.client_id == :office365 && event.host != cal_id
-      event = get_hosts_event(event)
-      event_id = event.id.not_nil!
+    # TODO:: instead of this we should store ical UID in the guest JWT
+    if client.client_id == :office365 && event.host != user_email
+      begin
+        event = get_hosts_event(event)
+        event_id = event.id.not_nil!
+      rescue PlaceCalendar::Exception
+        # we might not have access
+      end
```

Comment:
I assume not doing anything here is intentional?

Reply:
yep, I'll remove this extra query eventually